### PR TITLE
Test to compare DocumentSchema, SchemaProperty, SchemaDictProperty

### PR DESCRIPTION
**Pull request by kaapstorm**
@snopoke cc @emord

### CODE REVIEW

File: corehq/ex-submodules/dimagi/ext/tests/test_schema.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class SchemaTests(SimpleTestCase):
+    """
+    Demonstrates the difference between DocumentSchema, SchemaProperty
+    and SchemaDictProperty
+    """
+
+    def test_schemas(self):
+        spam = dict(Spam.wrap({}))
+        self.assertSetEqual(set(spam.keys()), {'doc_type', 'ham_dict_prop', 'ham_prop'})  # ham is not in spam
```
Comment (snopoke) [inline corehq/ex-submodules/dimagi/ext/tests/test_schema.py]: nit: don't need `.keys()`

File: corehq/ex-submodules/dimagi/ext/tests/test_schema.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class SchemaTests(SimpleTestCase):
+    """
+    Demonstrates the difference between DocumentSchema, SchemaProperty
+    and SchemaDictProperty
+    """
+
+    def test_schemas(self):
+        spam = dict(Spam.wrap({}))
```
Comment (snopoke) [inline corehq/ex-submodules/dimagi/ext/tests/test_schema.py]: nit: equivalent to `dict(Spam())`

**Review by snopoke — APPROVED** [review]
small nits but other wise :+1:

**Comment by kaapstorm** [comment]
I appreciate the nits. (Stickler also [taught me something](https://github.com/PyCQA/pylint/pull/1115) about \_\_eq\_\_ and Py3.)

**Pull request merged**